### ISSUE TITLE
Flux configuration prohibit creation of a new AKS cluster

### ISSUE THREAD

**Issue by ootneim** [Altinn/altinn-platform#866]
Terraform code for flux in platform prohibit creation of a new AKS cluster.

The resources:
- kubernetes_manifest.flux-slack-provider
- kubernetes_manifest.helm_repository_altinn_charts
- kubernetes_manifest.oci_repository_altinn_receipt
- kubernetes_manifest.kustomization_altinn_receipt

Terraform Plan gives "**cannot create REST client: no client config**" for these resources. Probably beacouse the AKS cluster in not provisoned during validation.

Workaround:

1. Deploy AKS cluster using tag **1.7.75**
2. Deploy AKS cluster again with tag **1.7.76**
3. Deploy AKS cluster again with a higer tag e.g. **1.7.86**

Destroy AKS cluster using Terraform is also broken. Currently only manual destroy is possible ( and to delete the state file)

**Comment by monteiro-renato** [comment]
Yes, we looked into this today and that's what the docs also warn about: 
[This resource requires API access during planning time. This means the cluster has to be accessible at plan time and thus cannot be created in the same apply operation. We recommend only using this resource for custom resources or resources not yet fully supported by the provider.](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest#before-you-use-this-resource).
These resources must be plan'ed and apply'ed after the cluster is created.

**Comment by monteiro-renato** [comment]
Using the [kubectl provider ](https://registry.terraform.io/providers/gavinbunney/kubectl/latest/docs/resources/kubectl_manifest)instead worked. We should not be in a broken state anymore (Well, as soon as we actually merge this into master).

I think it might be worth it to look into separating the setup of the infrastructure from the configuration of applications/services we run. Tweaking manifests via Terraform is not great in my experience.

**Comment by monteiro-renato** [comment]
We ended up having to add some more depends_on configs for other failures we saw while testing the original solution to the problem indicated in the issue description.

We saw issues with Pyrra resources failing to apply due to the CRDs not yet being present and with PodMonitor resources failing to apply due to the namespaces in which the resources were to be created not yet being available.

Another comment from @ootneim, is that the k8s setup stage now takes considerably longer to apply.
